intelmq/bots/collectors/file/collector_file.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""
+File Collector Bot
+
+Regularly checks a directory and reads files matching an extension from it
+It can delete the file if a delete switch was set.
+The bot creates on report per file.
+
+Copyright (C) 2016 by Bundesamt für Sicherheit in der Informationstechnik
+Software engineering by Intevation GmbH
+
+Parameters:
+path: string
+extension: string
+delete_file: boolean
+    default: False
+"""
+
+import sys
+import os
+import fnmatch
+
+from intelmq.lib.bot import Bot
+from intelmq.lib.message import Report
+
+
+class FileCollectorBot(Bot):
+
+    def init(self):
+        # Test if path is a directory
+        if not os.path.isdir(self.parameters.path):
+            self.logger.error("The path does not lead to a directory")
+            raise exceptions.InvalidArgument('path', got=self.parameters.path)
+
+        if not self.parameters.extension:
+            self.logger.warn("No file extension was set. The collector will" \
+                " read all files in %s", self.parameters.path)
+            if self.parameters.delete_file:
+                self.logger.error("This configuration would delete all files" \
+                    " in %s. I'm stopping now....", self.parameters.path)
+                self.stop()
+
+    def process(self):
+        self.logger.debug("Started looking for Files")
+
+        if os.path.isdir(self.parameters.path):
+            p = os.path.abspath(self.parameters.path)
+
+            #iterate over all files in dir
+            for f in os.listdir(p):
+                filename = os.path.join(p, f)
+                if os.path.isfile(filename):
+                    if fnmatch.fnmatch(f, '*'+self.parameters.extension):
```

Comment:
Best way would be the parameter. Adding a dot in this line would make matching on all files in this directory impossible